### DISCUSSION TITLE
Avoiding whole file import in TypeScript Cucumber Setup

### DISCUSSION THREAD

**Discussion by amalv** [cucumber/common#2178] [Q&A JavaScript] [answered]
Hello,

I'm wondering if there is a way to avoid importing the whole file like this in the steps:

```
import { add } from "../../src/index.ts";
```

If I remove the index.ts part, it shows me an error:

```
import { add } from "../../src";
```

My `tsconfig.json` file:

```
{
	"compilerOptions": {
		"module": "esnext",
		"moduleResolution": "node",
		"allowImportingTsExtensions": true,
		"noEmit": true
	}
}
```

`cucumber.json`:

```
{
	"default": {
		"loader": ["ts-node/esm"],
		"import": ["features/**/*.ts"]
	}
}
```

Error when running tests without `index.ts` and just importing `src` folder instead:

```
bun run test
$ cucumber-js
Error: ERR_UNSUPPORTED_DIR_IMPORT
```

**Comment by davidjgoss** [comment, marked as answer]
This is just how ESM works unfortunately, directory imports are not supported.
